src/mds/Server.cc

```diff
@@ -277,6 +277,18 @@ void Server::dispatch(const cref_t<Message> &m)
 {
   switch (m->get_type()) {
   case CEPH_MSG_CLIENT_RECONNECT:
+    if(mds->mdsmap->get_reject_client_session()) {
+      mds->clog->warn() << "client could not reconnect as" 
+                        << " file system flag reject_client_session is set";
+      dout(7) << "client cannot reconnect when file system flag"
+              << " reject_client_session is set"
+              << dendl;
+      auto reply = make_message<MClientSession>(CEPH_SESSION_CLOSE);
+      reply->metadata["error_string"] = "client cannot reconnect when file system flag" 
+                                        " reject_client_session is set";
+      mds->send_message(reply, m->get_connection());
+      return;
+    }
```

Comment:
Move this to `handle_client_reconnect` so you can dump the session to clog/dout. i.e. just before `if (!session->is_open()) {` in that method.

Reply:
Okay, I'll can move it to `handle_client_reconnect` but it does dump the session with current location too.

Reply:
done